Ship_Game/Universe/UniverseState.cs

```diff
@@ -432,6 +435,26 @@ public void AddSolarSystem(SolarSystem system)
             }
         }
 
+        public void AddResearchableSolarBody(ExplorableGameObject solarBody)
+        {
+            ResearchableSolarBodies[solarBody] = new HashSet<int>();
```

Comment:
This doesn't look terribly efficient. We could actually use a SmallBitSet for this, instead of a HashSet

Reply:
(the performance and memory usage difference should be 50x difference)

Reply:
We are talking about around 10 objects here on average. for the whole game